### DISCUSSION TITLE
Git clone E3SM-Project produces empty ${CODE_ROOT}/cime/ directory?

### DISCUSSION THREAD

**Discussion by jacob-stu-allen** [E3SM-Project/E3SM#5532] [E3SM model help]
Hello again!

Apologies for my many questions - 

In attempting to build my E3SMv2.1 case, I've noticed that the git clone of E3SM currently produces an empty ${CODE_ROOT}/cime/ directory. The run_e3sm.template.sh script attempts to find the cime create_newcase and xmlchange scripts here - but it's unable to find them. 

Am I just missing something here in regards to CIME? 

Thanks in advance,
Jacob

**Comment by jacob-stu-allen** [comment]
One quick update - when I also run  git submodule update --init --recursive, which cimes (ha) to grab the cime scripts, I get a permissions error. This is probably an issue with my ssh key, and may be irrelevant here. 

$ git submodule update --init --recursive

ERROR: 
Cloning into '/global/u2/j/jsallen/E3SMv2/code/20210806/cime'...
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

**Reply by xylar** [reply]
You will need to add your SSH key for the machine you are cloning into to your GitHub account:
https://docs.github.com/en/authentication/connecting-to-github-with-ssh/adding-a-new-ssh-key-to-your-github-account
If you haven't generated an SSH key yet, there is a link there to the instructions for generating one.
